From 50cba35f0ba469688a5f87c20f90e6ce4554bbcc Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 19 Apr 2024 15:02:02 +0100
Subject: [PATCH] ARM: dts: Add strict_gpiod dtparam

Setting strict_gpiod disables the GPIO output persistence, such that
pins are returned to being inputs when they are released. Note that
this applies to the GPIO/pinctrl driver for the user-facing GPIOs,
not the SoC GPIOs on Pi 5.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi | 2 ++
 arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi b/arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi
index 1d45fba0b392..f997d908e4b4 100644
--- a/arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm270x-rpi.dtsi
@@ -110,6 +110,8 @@ __overrides__ {
 		cam0_sync_inverted = <&csi0>, "sync-gpios:0=", <&gpio>,
 			    <&csi0>, "sync-gpios:4",
 			    <&csi0>, "sync-gpios:8=0", <GPIO_ACTIVE_LOW>;
+
+		strict_gpiod = <&chosen>, "bootargs=pinctrl_bcm2835.strict_gpiod=y";
 	};
 };
 
diff --git a/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi b/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi
index 359a30d6ef5f..89d12634f20b 100644
--- a/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -99,6 +99,7 @@ __overrides__ {
 		nvmem_cust_rw = <&nvmem_cust>,"rw?";
 		nvmem_priv_rw = <&nvmem_priv>,"rw?";
 		nvmem_mac_rw = <&nvmem_mac>,"rw?";
+		strict_gpiod = <&chosen>, "bootargs=pinctrl_rp1.strict_gpiod=y";
 	};
 };
 
-- 
2.34.1

